### PR TITLE
ASGI integration: return 400 Bad Request if json input malformed

### PR DESCRIPTION
**Pull request by Zlira**
## Description
Fixes returning "500 Internal Server Error" responses to requests with malformed JSON when running with ASGI integration.
This is described in the issue [#1214 ](https://github.com/strawberry-graphql/strawberry/issues/1214)
Django, Sanic and Flask integrations handle this scenario as expected. I've also added a test for Flask as it was missing. 

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* [#1214 ](https://github.com/strawberry-graphql/strawberry/issues/1214)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Fixes returning "500 Internal Server Error" responses to requests with malformed json when running with ASGI integration.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Olesia Grydzhuk for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: tests/asgi/test_query.py [hunk 1]
```diff
@@ -12,6 +12,13 @@ def test_simple_query(schema, test_client):
     assert response.json() == {"data": {"hello": "Hello world"}}
 
 
+def test_fails_when_request_body_has_invalid_json(test_client):
+    response = test_client.post(
+        "/", data='{"qeury": "{__typena"', headers={"content-type": "application/json"}
+    )
+    assert response.status_code == 400
```
Comment (BryceBeagle) [inline tests/asgi/test_query.py:19]: Not specific to this PR, but should we have a dedicated collection of tests for HTTP response codes? Is it worth it -- how many different situations are there?

**Review by BryceBeagle — APPROVED** [review]

**Review by patrick91 — APPROVED** [review]
Nice one! Thank you ☺️

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

File: tests/asgi/test_query.py (hunk 1 above)
Comment (DoctorJohn) [reply tests/asgi/test_query.py:19]: @BryceBeagle good point. @jkimbo mentioned in #1214 the GraphQL over HTTP spec. We should probably review our http views and make sure they comply with the spec and have sufficient tests, including tests response codes.

**Pull request merged**